### PR TITLE
chore: refactor Dockerfiles to reduce the build time

### PR DESCRIPTION
**Pull request by adriang-90**
Checklist:

* [c] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

## Summary
I managed to reduce build time by half with those changes. In addition I wanted to make a changes to Dockerfiles that follows the best practices, such as:
* Use **WORKDIR** instead of **RUN mkdir -p**
* Reduce number of layers
* Consolidate multiple **RUN** commands
* Add **yarn cache clean** -> https://github.com/hadolint/hadolint/issues/395
* Change **ADD** with **COPY** since you do not copy any *.tar files -> [nr 7](https://res.cloudinary.com/snyk/image/upload/v1551798390/Docker_Image_Security_Best_Practices_.pdf)
* Add [```--no-install-recommends```](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,13 +4,14 @@ ARG BASE_IMAGE=docker.io/library/ubuntu:21.10
...
-FROM docker.io/library/golang:1.18 as builder
+FROM docker.io/library/golang:1.18 AS builder
 
 RUN echo 'deb http://deb.debian.org/debian buster-backports main' >> /etc/apt/sources.list
 
-RUN apt-get update && apt-get install -y \
+RUN apt-get update && apt-get install --no-install-recommends -y \
     openssh-server \
     nginx \
+    unzip \
```
Comment (crenshaw-dev) [inline Dockerfile:14]: Why the extra dependency?

**Review by crenshaw-dev — COMMENTED** [review]
One clarifying question. Otherwise lgtm!

File: Dockerfile (hunk 1 above)
Comment (rishabh625) [reply Dockerfile:14]: may be bcoz ADD supports  auto extract of files and COPY doesn't

Comment (pasha-codefresh) [reply Dockerfile:14]: i have run it locally and without unzip we got

#20 21.71 + unzip /tmp/dl/awscli-linux-amd64-v2.4.6.zip -d /tmp/awscliv2
#20 21.71 ./installers/install-awscli-linux.sh: line 19: unzip: command not found

Comment (pasha-codefresh) [reply Dockerfile:14]: As i understand unzip was installed as part of apt-get install without no-install-recommends, and once we added it unzip stopped to be part of installation that cause problems above

Comment (adriang-90) [reply Dockerfile:14]: Sorry for 0 response till now, family issues. Yes as @pasha-codefresh stated above with no install recommends argument I couldn't unzip that's why I have added it. ADD got nothing to do with it afaik

**Review by pasha-codefresh — APPROVED** [review]
LGTM, thank you!

**Pull request merged**